src/polydb.jl

```diff
@@ -0,0 +1,301 @@
+   return Cursor{T}(Mongoc.find(c.mcol, Mongoc.BSON(d...)))
+end
+
+# creating `BSON` iterators from the respective `Polymake.BigObject` iterator
+function Collection{Mongoc.BSON}(c::Collection{Polymake.BigObject})
+   return Collection{Mongoc.BSON}(c.mcol)
+end
+
+function Cursor{Mongoc.BSON}(cursor::Cursor{Polymake.BigObject})
+   return Cursor{Mongoc.BSON}(cursor.mcursor)
+end
+
+# returns a Polymake.BigObject from a Mongoc.BSON document
+function parse_document(bson::Mongoc.BSON)
+   str = Mongoc.as_json(bson)
+   return call_function(:common, :deserialize_json_string, str)
+end
+
+#Iterator
+
+Base.IteratorSize(::Type{<:Cursor}) = Base.SizeUnknown()
+Base.eltype(::Cursor{T}) where T = T
+
+# help functions to reduce runtime and keep code clean
+function _evaluateIteration(nothing::Nothing)
+   return nothing
+end
+
+function _evaluateIteration(a::Tuple{Mongoc.BSON, Any})
+   return (parse_document(a[1]), a[2])
+end
+
+# default iteration functions returning `Polymake.BigObject`s
+function Base.iterate(cursor::Cursor{Polymake.BigObject})
+   return _evaluateIteration(iterate(cursor.mcursor))
+end
+
+function Base.iterate(cursor::Cursor{Polymake.BigObject}, state::Nothing)
+   return _evaluateIteration(iterate(cursor.mcursor, state))
+end
+
+function Base.iterate(coll::Collection{Polymake.BigObject})
+   return _evaluateIteration(iterate(coll.mcol))
+end
+
+function Base.iterate(coll::Collection{Polymake.BigObject}, state::Mongoc.Cursor)
+   return _evaluateIteration(iterate(coll.mcol, state))
+end
+
+# functions for `BSON` iteration
+function Base.iterate(cursor::Cursor{Mongoc.BSON})
+   return iterate(cursor.mcursor)
+end
+
+function Base.iterate(cursor::Cursor{Mongoc.BSON}, state::Nothing)
+   return iterate(cursor.mcursor, state)
+end
+
+function Base.collect(cursor::Cursor{Mongoc.BSON})
+   return collect(cursor.mcursor)
+end
+
+function Base.iterate(coll::Collection{Mongoc.BSON})
+   return iterate(coll.mcol)
+end
+
+function Base.iterate(coll::Collection{Mongoc.BSON}, state::Mongoc.Cursor)
+   return iterate(coll.mcol, state)
+end
+
+#Info
+
+# prints a list of the fields of a collection
+
+function fields(coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
+   info = collect(coll_c)[2]
+   if haskey(info, "fields")
+      return _get_fields(info["fields"])
+   else
+      return "no information on fields available"
+   end
+end
+
+function _get_fields(d::Dict)
+   res = ""
+   for (key, value) in d
+      if value == 1
+         res = string(res, key, "\n")
+      else
+         res = string(res, key, ": ", value, "\n")
+      end
+   end
+   return res
+end
+
+function _get_fields(a::Array)
+   return join(a, "\n")
+end
+
+# prints information about a specific Collection
+# also used for the info(::Database) function
+function info(coll::Collection)
+   db = coll.mcol.database
+   coll_c = db[string("_collectionInfo.", coll.mcol.name)]
```

Comment:
all string operations should happen in here (joining, filtering, etc.)

since we're printing something for the user to inspect visually, this function on the julia side should be rather called `Base.show(io::IO, coll::Collection)`
it also takes middle arguments of type MIME, have a look here: https://docs.julialang.org/en/v1/manual/types/#man-custom-pretty-printing-1